tests/models/components/test_seq2seq_encoder.py

```diff
@@ -0,0 +1,111 @@
+import pytest
+import torch
+
+from pie_models.models.components.seq2seq_encoder import (
+    ACTIVATION_TYPE2CLASS,
+    RNN_TYPE2CLASS,
+    build_seq2seq_encoder,
+)
+
+
+def test_no_encoder():
+    seq2seq_dict = {}
+    encoder, _ = build_seq2seq_encoder(seq2seq_dict, 10)
+    assert encoder is None
+
+    seq2seq_dict = {
+        "type": "sequential",
+        "rnn_layer": {
+            "type": "none",
+        },
+    }
+    encoder, _ = build_seq2seq_encoder(seq2seq_dict, 10)
+    assert len(encoder) == 0
+
+
+@pytest.mark.parametrize("seq2seq_enc_type", list(RNN_TYPE2CLASS))
+def test_rnn_encoder(seq2seq_enc_type):
+    seq2seq_dict = {
+        "type": "sequential",
+        "rnn_layer": {
+            "type": seq2seq_enc_type,
+            "hidden_size": 10,
+        },
+    }
+    encoder, _ = build_seq2seq_encoder(seq2seq_dict, 10)
```

Comment:
also check the second result value (also in tests below)

Reply:
Im doing the second value (output) tests in a separate function, thats why i'm ignoring it here